src/main/rx/spektrum.c

```diff
@@ -157,6 +158,147 @@ static int8_t dBm2range (int8_t dBm)
 }
 #endif
 
+ #ifdef USE_SPEKTRUM_VTX_CONTROL
+ #ifdef VTX_COMMON
+
+// Use the common set/get-frequncy API.
+// Much more generic than "knowing" someone elses indexes and trying to convert..
```

Comment:
Can all VTX types be set by frequency, or do some actually require setting by band / channel?

Reply:
There are no such limitations described. Also from what I can see in the vtx_common code.

Reply:
All known VTX devices registers their SetFreq function:
$ grep setFrequency -r src/main/
....
src/main/io/vtx_smartaudio.c:    .setFrequency = vtxSASetFreq,
src/main/io/vtx_tramp.c:    .setFrequency = vtxTrampSetFreq,
src/main/io/vtx_rtc6705.c:    .setFrequency = vtxRTC6705SetFreq,

I.e. a device independent generic vtx_common API. Wish there was a similar for setting power.

Reply:
```
void vtxRTC6705SetFreq(uint16_t freq)
{
    UNUSED(freq);
    return;
}
```

At least for RTC6705 this should be done by setting band / channel, instead of doing it in a way that will not work.

Reply:
OK, that is a problem on the output device end. This PR is on the input end. 
Easy fix is to undefine USE_VTX_COMMON_FREQ_API. Or make vtx_common a proper generic device independent API. Why must the spektrum Rx input device driver need to know all details on the VTX output devices? In this case there is a need to translate from spektrum band and channel indexes to vtx_common indexes. The code is there to do that, conditional on USE_VTX_COMMON_FREQ_API. But that is ugly. But then the power translation tables are also rather ugly. 
This all needs some work, on the output VTX device end.

Reply:
> Easy fix is to undefine USE_VTX_COMMON_FREQ_API.

Yes we should do that then.